Darken outline hover color to match default button hover

afholderman opened the pull request:
Alter button-outline-variant mixin to darken hover and active background in same fashion as filled button.

Fixes #24144

afholderman committed:
Fix Issue #24144

Alter button-outline-variant mixin to darken hover and active background in same fashion as filled button.

andresgalante commented on scss/mixins/_buttons.scss:
```diff
@@ -42,16 +42,16 @@
 }
 
-@mixin button-outline-variant($color, $color-hover: #fff) {
+@mixin button-outline-variant($color, $color-hover: #fff, $active-background: darken($color, 7.5%), $active-border: darken($color, 10%)) {
```
Since you are using the same parameters as in `button-variant` wouldn't it be better is we put outline variant under the same umbrella as normal variants? 
That way we keep them DRY and give less places to change to users.
What do you think?

andresgalante reviewed (approved):
Hi @afholderman I've pulled your branch, tested it, and it works as it should. Good job! 👍 

I am ok to merge this, but I'd keep in mind for the future to look for a better way to merge the `button-outline-variant` and `button-variant` mixins so we don't repeat the parameters. Set the colors once, once them in both places. specially the hover/active darken function.

afholderman committed:
Fix Issue #24144

Filled button on hover should utilize the same color-yiq mixin so that theme buttons match

afholderman commented:
@andresgalante I agree that these two mixins may need to be merged in to one. I just pushed another commit to fix another issue with the outline buttons, the lack of color-yiq usage. This was causing btn-light-outline to be illegible when hovered, and not match btn-light

afholderman committed:
Merge branch 'v4-dev' into button-outline-variant

mdo committed:
Merge branch 'v4-dev' into button-outline-variant

mdo committed:
default values, fix mixin
